components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -171,97 +202,125 @@ class NewTabPage extends React.Component<Props, State> {
       return null
     }
 
+    const hasImage = this.imageSource !== undefined
+    const isShowingBrandedWallpaper = newTabData.brandedWallpaperData ? true : false
+
     return (
       <App dataIsReady={newTabData.initialDataLoaded}>
         <PosterBackground
-          hasImage={newTabData.showBackgroundImage}
+          hasImage={hasImage}
           imageHasLoaded={this.state.backgroundHasLoaded}
         >
-          {newTabData.showBackgroundImage && newTabData.backgroundImage &&
-            <img src={newTabData.backgroundImage.source} />
+          {hasImage &&
+            <img src={this.imageSource} />
           }
         </PosterBackground>
-        {newTabData.showBackgroundImage &&
+        {hasImage &&
           <Gradient
             imageHasLoaded={this.state.backgroundHasLoaded}
           />
         }
         <Page>
-          <Header>
+          {newTabData.showStats &&
+          <S.GridItemStats>
             <Stats
               textDirection={newTabData.textDirection}
               stats={newTabData.stats}
-              showWidget={newTabData.showStats}
               hideWidget={this.toggleShowStats}
               menuPosition={'right'}
             />
+          </S.GridItemStats>
+          }
+          {newTabData.showClock &&
+          <S.GridItemClock>
             <Clock
               textDirection={newTabData.textDirection}
-              showWidget={newTabData.showClock}
               hideWidget={this.toggleShowClock}
               menuPosition={'left'}
             />
+          </S.GridItemClock>
+          }
+          {this.props.newTabData.gridSites.length && newTabData.showTopSites &&
```

Comment:
React is rendering a `0` here and top sites are not visible. Reproducible in a fresh profile after visiting a few websites. No flags used

Reply:
after enabling the flag I was able to see top sites immediately. after relaunching without flags top sites persisted so it seems related to a fresh profile condition.

Reply:
Good spot - I'll change to `boolean` or `undefined`

Reply:
This should be fixed.